Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -59,7 +59,7 @@ class LoginsHelper: BrowserHelper {
                    let requestId = res["requestId"] as? String {
                     requestLogins(login, requestId: requestId)
                 }
-            } else if type == "submit" {
+            } else if type == "submit" && self.profile.prefs.boolForKey("saveLogins") ?? true {
```

Comment:
nit: I prefer to have the second conditional in brackets. Makes it easier to read and much clearer.

`&& (self.profile.prefs.boolForKey("saveLogins") ?? true)`